interpreter/text/parser.mly

```diff
@@ -39,6 +39,45 @@ let ati i =
 let literal f s =
   try f s with Failure _ -> error s.at "constant out of range"
 
+let range_check i min max at =
+    let i_32 = Int32.to_int i in
+    let min_32 = Int32.of_int min in
+    let max_32 = Int32.of_int max in
+    if i > max_32 || i < min_32 then error at "constant out of range" else i_32
+
+let int8_of_string s = range_check (I32.of_string s.it) (-128) 255 s.at
+let int16_of_string s = range_check (I32.of_string s.it) (-32768) 65535 s.at
+let int32_of_string s = I32.to_bits (literal (fun s -> I32.of_string s.it) s)
+let int64_of_string s = I64.to_bits (literal (fun s -> I64.of_string s.it) s)
+let f32_of_string s = F32.to_bits (literal (fun s -> F32.of_string s.it) s)
+let f64_of_string s = F64.to_bits (literal (fun s -> F64.of_string s.it) s)
+
+let simd_literal f shape ss =
```

Comment:
The `f` parameter seems to be unused? How do you detect, e.g., `(i32.const f64x2 0 0)`?

To avoid that, it may be easiest to turn `v128.const` into a separate keyword.

Reply:
Separate v128.const into a separate keyword, thanks for the suggestion!